cli/src/main/java/com/devonfw/tools/ide/tool/yarn/Yarn.java

```diff
@@ -0,0 +1,53 @@
+package com.devonfw.tools.ide.tool.yarn;
+
+import java.nio.file.Path;
+import java.util.Set;
+
+import com.devonfw.tools.ide.common.Tag;
+import com.devonfw.tools.ide.context.IdeContext;
+import com.devonfw.tools.ide.process.ProcessContext;
+import com.devonfw.tools.ide.tool.npm.NpmBasedCommandlet;
+import com.devonfw.tools.ide.tool.repository.ToolRepository;
+import com.devonfw.tools.ide.version.VersionIdentifier;
+
+/**
+ * {@link NpmBasedCommandlet} for <a href="https://yarnpkg.com">yarn</a>.
+ */
+public class Yarn extends NpmBasedCommandlet {
+
+  /**
+   * The constructor.
+   *
+   * @param context the {@link IdeContext}.
+   */
+  public Yarn(IdeContext context) {
+
+    super(context, "yarn", Set.of(Tag.TYPE_SCRIPT, Tag.BUILD));
+  }
+
+  @Override
+  public String getInstalledEdition() {
+
+    if (hasNodeBinary(getName())) {
+      return "yarn";
+    }
+    return null;
+  }
+
+  @Override
+  protected void performToolInstallation(ToolRepository toolRepository, VersionIdentifier resolvedVersion, Path installationPath, String edition,
+      ProcessContext processContext) {
+    runPackageInstall("yarn@" + resolvedVersion);
+  }
```

Comment:
Why are you overriding this method? You are doing the same thing but forget to invalidate the cached value.
```suggestion
```